src/nunit.analyzers/TestCaseUsage/TestCaseUsageAnalyzer.cs

```diff
@@ -49,6 +49,9 @@ private static void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
                     var attributeSymbol = context.SemanticModel.GetSymbolInfo(attributeNode).Symbol;
                     var testCaseType = context.SemanticModel.Compilation.GetTypeByMetadataName(NunitFrameworkConstants.FullNameOfTypeTestCaseAttribute);
 
+                    if (testCaseType == null)
```

Comment:
Thanks. I'll make a subsequent commit. To be honest I have not thought much about performance yet, and I don't have much knowledge about the performance impact of the different methods in Roslyn. My plan was to do some profiling later when we are further in the project.

Reply:
As a rule of thumb we want to check what we can at syntax level before touching symbols. No need to go overboard with it though. In this case it would be pretty weird to have the analyzer installed but no reference to nunit so should be rare.